src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -2757,7 +2757,14 @@ public override InterfaceMapping GetInterfaceMap([DynamicallyAccessedMembers(Dyn
                 MethodBase? rtTypeMethodBase = GetMethodBase(reflectedType, classRtMethodHandle);
                 // a class may not implement all the methods of an interface (abstract class) so null is a valid value
                 Debug.Assert(rtTypeMethodBase is null || rtTypeMethodBase is RuntimeMethodInfo);
-                im.TargetMethods[i] = (MethodInfo)rtTypeMethodBase!;
+                RuntimeMethodInfo? targetMethod = rtTypeMethodBase as RuntimeMethodInfo;
```

Comment:
```suggestion
                RuntimeMethodInfo? targetMethod = (RuntimeMethodInfo)rtTypeMethodBase;
```
This is not expected to fail